zhaquirks/xiaomi/aqara/magnet_ac01.py

```diff
@@ -0,0 +1,75 @@
+"""Xiaomi aqara P1 contact sensor device."""
+
+from zigpy.profiles import zha
+import zigpy.types as t
+from zigpy.zcl.clusters.general import Basic, Identify, OnOff, Ota, PowerConfiguration
+from zigpy.zcl.clusters.security import IasZone
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.xiaomi import (
+    LUMI,
+    BasicCluster,
+    XiaomiAqaraE1Cluster,
+    XiaomiCustomDevice,
+)
+
+
+class OppleCluster(XiaomiAqaraE1Cluster):
+    """Opple cluster."""
+
+    class DetectionDistance(t.enum8):
+        """Detection distance."""
+
+        TenMM = 0x01
+        TwentyMM = 0x02
+        ThirtyMM = 0x03
+
+    attributes = {
+        0x010C: ("detection_distance", DetectionDistance, True),
+    }
+
+
+class LumiMagnetAC01(XiaomiCustomDevice):
+    """Custom device representing lumi.magnet.ac01."""
+
```

Comment:
This device seems to use a CR123A, so "battery size = 8" should be set:
(default is 10 / CR2032 for `XiaomiCustomDevice`)

```python
    def __init__(self, *args, **kwargs):
        """Init."""
        self.battery_size = 8
        super().__init__(*args, **kwargs)
```

Reply:
fixed